### ISSUE TITLE
Outdoor segmentation

### ISSUE THREAD

**Issue by Cureser** [Pointcept/Concerto#3]
I tried using the `concerto_large_outdoor` model with a segmentation head trained on the KITTI dataset to achieve semantic segmentation, but the results don't seem to be good. Would you like to know if there are any examples of outdoor point cloud semantic segmentation in the future? I would also like to know which datasets the `concerto_large_outdoor` pre-trained model uses. Thank you.

This is the configuration file I used for training.
`_base_ = ["../_base_/default_runtime.py"]

# 自定义设置
batch_size = 6  # 所有GPU的总批量大小
num_worker = 8
mix_prob = 0.5
clip_grad = 3.0
empty_cache = True
enable_amp = True
gradient_accumulation_steps = 4
weight = "../Concerto/model/concerto_large_outdoor.pth"
# 模型设置
model = dict(
    type="DefaultSegmentorV2",
    num_classes=19,  # 户外数据集的类别数
    backbone_out_channels=1728,  # Concerto_large的输出特征维度
    backbone=dict(
        type="PT-v3m2",
        in_channels=4,  # xyz + normal
        order=("z", "z-trans", "hilbert", "hilbert-trans"),
        stride=(2, 2, 2, 2),
        enc_depths=(3, 3, 3, 12, 3),
        enc_channels=(64, 128, 256, 512, 768),
        enc_num_head=(4, 8, 16, 32, 48),
        enc_patch_size=(1024, 1024, 1024, 1024, 1024),
        mlp_ratio=4,
        qkv_bias=True,
        qk_scale=None,
        attn_drop=0.0,
        proj_drop=0.0,
        drop_path=0.3,
        shuffle_orders=True,
        pre_norm=True,
        enable_rpe=False,
        enable_flash=True,
        upcast_attention=False,
        upcast_softmax=False,
        traceable=False,
        mask_token=False,
        enc_mode=True,
        freeze_encoder=False,
    ),
    criteria=[
        dict(type="CrossEntropyLoss", loss_weight=1.0, ignore_index=-1),
        dict(type="LovaszLoss", mode="multiclass", loss_weight=1.0, ignore_index=-1),
    ],
    freeze_backbone=True,  # 冻结骨干网络，只训练分割头
)

# 训练调度器设置
# scheduler settings
epoch = 50
eval_epoch = 50
optimizer = dict(type="AdamW", lr=0.002, weight_decay=0.005)
scheduler = dict(
    type="OneCycleLR",
    max_lr=[0.002, 0.0002],
    pct_start=0.04,
    anneal_strategy="cos",
    div_factor=10.0,
    final_div_factor=100.0,
)
param_dicts = [dict(keyword="block", lr=0.0002)]

# dataset settings
dataset_type = "SemanticKITTIDataset"
data_root = "data/semantic_kitti"
ignore_index = -1
names = [
    "car",
    "bicycle",
    "motorcycle",
    "truck",
    "other-vehicle",
    "person",
    "bicyclist",
    "motorcyclist",
    "road",
    "parking",
    "sidewalk",
    "other-ground",
    "building",
    "fence",
    "vegetation",
    "trunk",
    "terrain",
    "pole",
    "traffic-sign",
]

data = dict(
    num_classes=19,
    ignore_index=ignore_index,
    names=names,
    train=dict(
        type=dataset_type,
        split="train",
        data_root=data_root,
        transform=[
            dict(type="RandomRotate", angle=[-1, 1], axis="z", center=[0, 0, 0], p=0.5),
            dict(type="RandomScale", scale=[0.9, 1.1]),
            dict(type="RandomFlip", p=0.5),
            dict(type="RandomJitter", sigma=0.005, clip=0.02),
            dict(
                type="GridSample",
                grid_size=0.05,
                hash_type="fnv",
                mode="train",
                return_grid_coord=True,
            ),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
        ignore_index=ignore_index,
    ),
    val=dict(
        type=dataset_type,
        split="val",
        data_root=data_root,
        transform=[
            dict(type="Copy", keys_dict={"segment": "origin_segment"}),
            dict(
                type="GridSample",
                grid_size=0.05,
                hash_type="fnv",
                mode="train",
                return_grid_coord=True,
                return_inverse=True,
            ),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment", "origin_segment", "inverse"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
        ignore_index=ignore_index,
    ),
    test=dict(
        type=dataset_type,
        split="test",
        data_root=data_root,
        transform=[
            dict(type="Copy", keys_dict={"segment": "origin_segment"}),
            dict(
                type="GridSample",
                grid_size=0.025,
                hash_type="fnv",
                mode="train",
                return_inverse=True,
            ),
        ],
        test_mode=True,
        test_cfg=dict(
            voxelize=dict(
                type="GridSample",
                grid_size=0.05,
                hash_type="fnv",
                mode="test",
                return_grid_coord=True,
            ),
            crop=None,
            post_transform=[
                dict(type="ToTensor"),
                dict(
                    type="Collect",
                    keys=("coord", "grid_coord", "index"),
                    feat_keys=("coord", "strength"),
                ),
            ],
        ),
    ),
)

# runner settings
runner = dict(
    type="Runner",
    max_epoch=epoch,
    train_dataloader=dict(
        batch_size=batch_size,
        num_workers=8,
        persistent_workers=True,
        pin_memory=True,
    ),
    val_dataloader=dict(
        batch_size=batch_size,
        num_workers=8,
        persistent_workers=True,
        pin_memory=True,
    ),
    test_dataloader=dict(
        batch_size=batch_size,
        num_workers=8,
        persistent_workers=True,
        pin_memory=True,
    ),
    auto_resume=True,
    checkpointsave_cfg=dict(
        interval=10,
        max_keep_ckpts=1,
    ),
    log_cfg=dict(
        interval=50,
        hooks=[
            dict(type="TextLoggerHook"),
            dict(type="TensorboardLoggerHook"),
        ],
    ),
)`

**Comment by Yujia-Zhang0913** [comment]
Hi, in your config, it seems that you use `in_channels=4, # xyz + normal`, I think the model may not be correctly loaded. Because the model is trained on [coord, color, normal], which means the in_channels is 9. However, due to the data sometimes does not have color or normal, you can try to set the color to black, and the normal can be set to zeros or the normalized vector in the direction from the point to the lidar.

Could you share the train.log? I can check whether the weight is correctly loaded.

**Comment by Cureser** [comment]
Changing in_channels to 9 still doesn't load the model weights correctly. Do we need to do any additional configuration or operation on the loaded concerto_large_outdoor.pth model?

**Comment by Yujia-Zhang0913** [comment]
Hi, I have updated some outdoor eval configs. For KITTI, you can use [semseg-ptv3-large-v1m1-kitti-4a-lin.py](https://github.com/Pointcept/Pointcept/blob/main/configs/concerto/semseg-ptv3-large-v1m1-kitti-4a-lin.py) for point clouds without color and normal and [semseg-ptv3-large-v1m1-kitti-4a-withcolornormal-lin.py](https://github.com/Pointcept/Pointcept/blob/main/configs/concerto/semseg-ptv3-large-v1m1-kitti-4a-withcolornormal-lin.py) for point clouds with color from images and normal from estimation.

When using `concerto_large_outdoor.pth`, please replace these lines:
```
dict(
        type="CheckpointLoader",
        keywords="module.student.backbone",
        replacement="module.backbone",
    ),
```
to 
```
dict(
        type="CheckpointLoader",
        keywords="module",
        replacement="module.backbone",
    ),
```
to correctly load the weight.

If you have any further issues, feel free to ask.

**Comment by Yujia-Zhang0913** [comment]
The author has successfully loaded the weight and gotten results in #5. This issue will be closed.
